System.Containers.Tasks/CreateNewImage.cs

```diff
@@ -1,105 +1,146 @@
-            {
-                Log.LogError("PublishDirectory and Files are both invalid. One valid parameter MUST be given to the CreateNewImage task.");
-                return false;
-            }
+            Log.LogError(string.Format("PublishDirectory is must have a value"));
+            return !Log.HasLoggedErrors;
+        }
+        else if (!Directory.Exists(PublishDirectory))
+        {
+            Log.LogError(string.Format("PublishDirectory does not exist."));
+            return !Log.HasLoggedErrors;
+        }
 
-            Registry reg = new Registry(new Uri(InputRegistryURL));
+        Registry reg;
+        Image image;
 
-            Image image;
-            try
-            {
-                image = reg.GetImageManifest(BaseImageName, BaseImageTag).Result;
-            }
-            catch (Exception ex)
+        try
+        {
+            reg = new Registry(new Uri(BaseRegistry, UriKind.RelativeOrAbsolute));
+            image = reg.GetImageManifest(BaseImageName, BaseImageTag).Result;
+        }
+        catch (Exception e)
+        {
+            if (BuildEngine != null)
             {
-                Log.LogError("GetImageManifest Failed: {0}.\n{1}", ex.Message, ex.InnerException);
-                return false;
+                Log.LogError("Failed getting image manifest: {0}", e);
             }
+            return !Log.HasLoggedErrors;
+        }
 
+        if (BuildEngine != null)
+        {
             Log.LogMessage($"Loading from directory: {PublishDirectory}");
-            Layer newLayer = Layer.FromDirectory(PublishDirectory, WorkingDirectory);
-            image.AddLayer(newLayer);
-
-            image.SetEntrypoint(Entrypoint, EntrypointArgs?.Split(' ').ToArray());
-
-            Registry outputReg = new Registry(new Uri(OutputRegistryURL));
+        }
+        
+        Layer newLayer = Layer.FromDirectory(PublishDirectory, WorkingDirectory);
+        image.AddLayer(newLayer);
+        image.SetEntrypoint(Entrypoint, EntrypointArgs?.Split(' ').ToArray());
 
+        if (OutputRegistry.StartsWith("docker://"))
+        {
+            // To Do: LocalDocker.Load();
+        }
+        else
+        {
+            Registry outputReg = new Registry(new Uri(OutputRegistry));
             try
             {
-                outputReg.Push(image, NewImageName, BaseImageName).Wait();
+                outputReg.Push(image, ImageName, BaseImageName).Wait();
             }
             catch (Exception e)
             {
-                Log.LogError("Failed to push to the output registry: {0}\n{1}", e.Message, e.InnerException);
-                return false;
+                if (BuildEngine != null)
+                {
+                    Log.LogError("Failed to push to the output registry: {0}", e);
+                }
+                return !Log.HasLoggedErrors;
             }
-
-            return true;
         }
+
+        return !Log.HasLoggedErrors;
```

Comment:
I think in the success case there should be a high-pri log message along the lines of

```
Pushed container 'myorg/myapp:1.2.3' to registry 'myorgcontainers.azurecr.io'
```